Recipe for diffpy.srfit package

sbillinge opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/diffpy.srfit/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/diffpy.srfit/meta.yaml**:

 * ℹ️ License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15944775990](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15944775990). Examine the logs at this URL for more detail.</sub>

sbillinge commented:
@cadenmyers13, @Tieqiong and @dragonyanglong please can you comment on this PR if you are willing to listed as a conda-forge maintainer for this package.  Thanks.

sbillinge commented:
@tieqiong @cadenmyers13 I am still working on this please hold off review for the time being

sbillinge commented:
> By the way is this doc url correct?

it should be but this is a decision we have to make.  Currently `diffpy.github.io` redirects to `diffpy.org`.  Whether we want this to be the case in the future we should think about it....

ocefpaf commented on recipes/diffpy.srfit/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+{% set name = "diffpy.srfit" %}
+{% set version = "3.1.0rc0" %}
```
This one falls under https://github.com/conda-forge/cfep/blob/main/cfep-05.md

sbillinge replied:
thanks for the link.  Very helpful.

sbillinge replied:
It is not completely clear to me how we do this labelling in the staged-recipe.   To make things easier I think we could wait for our full release to appear on PyPI and then change this to a non-rc release.   I guess merging to an `rc` branch as described in the article by Eric is possible from the feedstock but not hte staged recipe?  Anyway, this is easily fixed if we skip the rc release to conda-forge this time.

however, please see below for the licensing issue.....

sbillinge replied:
I think this is better also.  We will work on that and loop back when it is done.  

We wanted to do it this way as we wanted to conda install this into another package and wanted to test everything including the conda imports before it is public, but since noone is using it as it doesn't exist on CF yet, I think it is just easier to push up a full release.

sbillinge replied:
sorry, that was ambigous I see when I reread, what I mean is that we will make a stable release instead of the rc

ocefpaf commented on recipes/diffpy.srfit/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+
+about:
+  home: https://github.com/diffpy/diffpy.srfit/
+  license: Free To Use But Restricted
+  license_file: LICENSE.rst
```
Custom license make it very hard for us to accept b/c we are not lawyers and we cannot evaluate all the possible interpretations and consequences. Can ask upstream to if they can use a well-known OSI approved license instead?

ocefpaf replied:
I'm not a lawyer,  and sometimes recipes get merged without the proper license care. If you write "this is 100% compatible with bsd3" I would not mind merging it as-is. I will refer to your comment though if any questions arise. Is that ok?

PS: if you can get upstream to move to bsd3 it would be awesome!

sbillinge commented:
@conda-forge/help-python, @ocefpaf, this is ready for review!

We have addressed all the comments from the previous reviews.

The one last question is about the non-standard license.  We would like to request that this gets merged in this form.  We note that the license has all the conditions of the standard BSD-3 clause license, on which it is based, but there is simply an additional condition that the US government reserves a no-cost non exclusive right to use the software.   Actually, this is ensured by the BSD-3 clause so does not change the terms from there.  We also note that we have a number of packages already on conda-forge with this exact license: diffpy.utils, diffpy.pdfgui and diffpy.structure.

We will work in the future with Brookhaven National Lab and see if we can return the license to a standard BSD-3, but this could take some time and if we can't get this package (`diffpy.srfit`) and another package we are working on, `diffpy.srreal` that has the same license, it will be very difficult to release the whole materials structure package `diffpy.cmi`., which is top priority for us.

Please could you consider merging this (and diffpy.srreal when it is ready) with the current version of the open-source license?

Thanks very much for the consideration.

ocefpaf commented on recipes/diffpy.srfit/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+
+about:
+  home: https://github.com/diffpy/diffpy.srfit/
+  license: Free To Use But Restricted
```
Can you use `LicenseRef-diffpy` instead? It would be nice to standardized the others onez that are already in conda-forge as well to make then easily discoverable.

sbillinge replied:
I have changed this to `LicenseRef-diffpy` as requested and added a statement that the license file is 100% compatible with BSD-3 as you requested.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/diffpy.srfit/meta.yaml```) and found it was in an excellent condition.

sbillinge commented:
btw, @ocefpaf I will take your advice and make these changes in each of our affected packages so they are all consistent.
